custom zkticket proof screen

saurfang opened the pull request:
fixes https://github.com/proofcarryingdata/zupass/issues/963

* custom proof screen for zk ticket
    * had to implement eligible ticket check explicitly 
* conditionally show custom logo for ZuKat, can be extended for more generic usage in the future
* custom handling of
    * valid event ids: show event names passed via argument description for now (can be extended as argument types in the future) 
    I decide to pass explicitly with argument to avoid creating new routes on prove screen to fetch event names async. This will involve additional auth and rate limiting consideration.
    * semaphore identity
    * watermark w/ and w/o human readable payload (e.g. anon message)
    * other fields to reveal
* fallback to hidden PCDArgs for unhandled args
* remove `hideIcon` option in Argument which is no longer used/needed

tested on consumer client examples

<img width="429" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/5140da9c-1674-40f2-a013-7f9e2dbb41c1">

anon msg send / long msg are collapsed by default

<img width="416" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/6010ec37-f5bd-45d8-9671-ee6924416798">

click to expand up to 4 lines and scrollable

<img width="406" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/937c10a5-4e05-4990-9da6-802a45f68a89">

tg auth

<img width="408" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/6d7bbe79-901b-4f21-b666-006950be36a3">

show hidden args

<img width="407" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/e530ed98-0205-4636-9c43-d4b7f5a75fde">

no ticket found

<img width="411" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/f5433b49-0344-4296-b26c-741fc72d331a">

lermchair commented:
can we also change the contrast and/or color of the "not revealed" fields instead of just the icon?

saurfang commented:
updated color a bit. also fixed a bug where not revealed fields were not showing in the previous screenshots

<img width="451" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/9a62b119-0065-44de-9d3e-9c03e1cb6679">

saurfang commented:
restyled message container with background color

<img width="559" alt="image" src="https://github.com/proofcarryingdata/zupass/assets/4317392/33aa5ada-edfe-46a0-979f-1a3013d7a6ab">

rrrliu commented:
Closing for now as low-priority
